### ISSUE TITLE
Set aspect ratio to 16:9 during cutscenes

### ISSUE THREAD

**Issue by InterClaw** [fayaz12g/totk-aar#17]
Would it be possible to temporarily set aspect ratio to 16:9 during cutscenes? And then switch it back to whatever it was when the cutscene ends (e.g. stretched). I guess the same detection that is used in DynamicFPS/DynamicFPS++ can be used for this.

**Comment by fayaz12g** [comment]
Theoretically yes, I will look into doing this once I have added the other features I’ve promised

**Comment by xNicklaj** [comment]
If it can help you in any way I'm quite sure that the original theboy 21:9 mod had this feature, maybe you can give it a look and adapt it for any resolution

**Comment by fayaz12g** [comment]
@xNicklaj I'll take a look at it, do you have a copy of it anywhere so I can test it? Also I appreciate all the UI issues you've found, would you be willing to look into helping me make fixes? I'm also a student learning all this haha

**Comment by xNicklaj** [comment]
I would honestly not dislike it, the problem is that I have no idea whatsoever on how to start or where to start. I don't despise the idea of learning from scratch though!

Hit me up on Discord (user: nicklaj) and we can talk it through!

**Comment by Fruithapje21** [comment]
Let me clear up some things. Setting the aspect ratio is done by letting Yuzu scale the final image to fit your screen. The game itself will always output frames with a 16:9 aspect ratio. You cannot 'temporarily set the aspect to 16:9' because you didn't change this aspect in the first place. There is currently no way to change the aspect ratio setting of yuzu on the fly. The only thing you can do is setting the aspect ratio to 'stretch to window' and rescaling the window yourself when the cutscene is playing.

I also had a look at theboy's 21:9 mod. It sets the playback size of the prerendered video to 1280x900 (normally 1280x720). What this does is stretch the video vertically. However, you can still only see the top 720 pixels, so you are effectively watching a zoomed in version of the video. If you were to extend this approach to larger aspect ratios, you will zoom in more and more and see less and less of the original video. 

If you, instead of stretching he video vertically, squish the video horizontally, you _do_ end up with the correct aspect ratio _and_  will be able see the full video, _but_ the remaining pixels will be read from out of bounds memory. For 21:9 I ended up with this masterpiece
![Screenshot 2023-07-17 234405](https://github.com/fayaz12g/totk-aar/assets/136446038/1edf561d-4318-4334-bfa1-356776e2ae88)

**Comment by InterClaw** [comment]
I guess this means the only known way of getting effective 16:9 video in a 21:9 viewport is to reencode the video with pillarboxing then?

**Comment by Fruithapje21** [comment]
Yes, I'm afraid so. Also good luck trying to distribute those without getting unwanted attention from Nintendo.

**Comment by xNicklaj** [comment]
You could use an approach where you don't distribute the encoded version, but rather a script that given the original files re-encodes them
